parser/src/java/org/apache/hadoop/hive/ql/parse/IdentifiersParser.g

```diff
@@ -988,10 +988,32 @@ nonReserved
     | KW_TAG
     | KW_FAST_FORWARD
     | KW_OPTIMIZE
+    | KW_APPLICATION
+    | KW_COMPACT_ID
+    | KW_CONF
+    | KW_DATACONNECTOR
+    | KW_DATACONNECTORS
+    | KW_DATABASE
+    | KW_DDL
+    | KW_EXCHANGE
+    | KW_EXTENDED
+    | KW_FOLLOWING
+    | KW_FORCE
+    | KW_IF
+    | KW_LESS
+    | KW_MACRO
+    | KW_MORE
+    | KW_OLDER
+    | KW_PKFK_JOIN
+    | KW_PRECEDING
+    | KW_SYNC
+    | KW_THAN
+    | KW_TIMESTAMPLOCALTZ
+    | KW_UNBOUNDED
 ;
```

Comment:
The current commit maximized the number of non-reserved words. But I think some of them(or all?) are intentionally configured. I guess we would like to revert some of them.
https://cwiki.apache.org/confluence/display/hive/languagemanual+ddl#LanguageManualDDL-Keywords,Non-reservedKeywordsandReservedKeywords

Just for convenience,

- Introduced by 1.2.0
    - CONF, DATABASE, EXCHANGE, EXTENDED, FOLLOWING, IF, LESS, MACRO, MORE, PRECEDING, UNBOUNDED
- Introduced by 3.0.0
    - SYNC
- Not mentioned in the document
    - APPLICATION, COMPACTIONID, CONNECTOR, CONNECTORS, DDL, FORCE, OLDER, PKFK_JOIN, THAN, TIMESTAMPLOCALTZ